### DISCUSSION TITLE
Allow pluggables (SFP, QSFP, ...) in front- and rear- ports

### DISCUSSION THREAD

**Discussion by netixx** [netbox-community/netbox#9459] [Ideas]
I opened issue #9400 but I am not sure I understand the answer.

Our teams needs to be able to use the front/rear-port logic (i.e. the binding between the two that allow end-to-end visibility on a path) but at the same time, we need to specify the interface type (so that we know what type of connectivity is provided (100G, 400G...).

Why would this be a bad idea ?

Code wise, it would only require to add a few lines in this class: https://github.com/netbox-community/netbox/blob/17292324a35465ff9343442ae0bb735c4a5d4ab5/netbox/dcim/choices.py#L1010

**Comment by Kludge-As-A-Service** [comment]
Not responding directly to your feature request, but if your need is end-to-end circuit visibility when passing through 2 interfaces on a device,  wouldn't a bridge do the trick?

> Multiple interfaces can be bridged to a single virtual interface to effect a bridge group. **Alternatively, two physical interfaces can be bridged to one another, to effect an internal cross-connect.**

https://docs.netbox.dev/en/stable/release-notes/version-3.1/#interface-bridging-6346

**Reply by netixx** [reply]
Bridge could solve the issue for one-to-one binding (even if it doesn't necessarily model what is going on).

In my case however, we need one-to-many associations, i.e. bind 4 "client" interfaces to 1 "network" interface in a unidirectional manner (the 4 client interfaces cannot see each other). This is the same as DWDM optical multiplexing (which is usually done through passive filters, which present LC front-ports and a LC rear-port), but in this case, we have active modules doing TDM (that could be modeled using QSFP front-ports and QSFP rear-port).

**Reply by DanSheps** [reply]
You need to wait until 3.3 which updates the cable module a bit to allow for duplex (and other types) of connections.

**Reply by netixx** [reply]
Looking forward to this! 
At the moment, we planned to use map the client ports as front-ports and bind them to the same rear port using an id (multiplex 4 front-ports to one rear-ports). This is actually a pretty good model of what is happening (except we cannot put the correct types for the interfaces).

**Comment by matejv** [comment]
If you click on a front or rear port there is an option to "Add inventory item" (see [here for example](https://demo.netbox.dev/dcim/front-ports/1300/)). Inventory items in netbox are meant to hold information about SFP pluggables and other filed replaceable modules. See [documentation on inventory items](https://demo.netbox.dev/static/docs/core-functionality/devices/#inventory-items) for more info.

**Reply by netixx** [reply]
I wasn't super familiar with binding inventory items to interfaces/front-ports/rear-ports. I guess this could circumvent the issue of storing the information about which pluggable is installed.

However, I would still get a wrong port type displayed in this view: https://demo.netbox.dev/dcim/devices/88/front-ports/ (and I would have to dig into the front-port detailed view to get the information I need).

In my opinion, having QSFP port type would be the better option (and we would use inventory items to store details about the pluggable, eg. serial numbers or part numbers).

**Reply by AiYoriAoshi** [reply]
I second @netixx opinion here. 

As I am totally new to netbox maybe I am modelling it wrong but this DWDM Box has 1 LC Port as Line which I modeled as Rear Port and 12 QSFP28 Slots which I modeled as Front Ports since I want to be able to trace the connection through the device, as it's not an end device.

I couldn't find any way to link an Interface to a rear port because that's not what an Interface is supposed to be. Interface is a termination port for a link. Sure, the data will take another interface continue but that's not netbox's business.

But for DWD that slot is not a termination, so it cannot be an Interface. That leaves only the Front Port - Rear Port way and therefore we need to be able to set to a slot type.
